build-logic/basics/src/main/kotlin/gradlebuild/basics/kotlindsl/kotlin-compiler-configuration.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.basics.kotlindsl
+
+import org.gradle.kotlin.dsl.*
+import org.jetbrains.kotlin.config.AnalysisFlags
+import org.jetbrains.kotlin.config.ApiVersion
+import org.jetbrains.kotlin.config.CommonConfigurationKeys
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.config.JVMConfigurationKeys
+import org.jetbrains.kotlin.config.JvmAnalysisFlags
+import org.jetbrains.kotlin.config.JvmClosureGenerationScheme
+import org.jetbrains.kotlin.config.LanguageVersion
+import org.jetbrains.kotlin.config.LanguageVersionSettingsImpl
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.dsl.KotlinVersion
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import org.jetbrains.kotlin.load.java.JavaTypeEnhancementState
+import org.jetbrains.kotlin.load.java.Jsr305Settings
+import org.jetbrains.kotlin.load.java.ReportLevel
+
+
+fun KotlinCompile.configureKotlinCompilerForGradleBuild() {
+    compilerOptions {
+        allWarningsAsErrors = true
+        apiVersion = KotlinVersion.KOTLIN_1_8
```

Comment:
🤔 Is there a way we could share the common values between the two extension methods?

Right now, this code remains problematic as someone could update a line in one function and not in the other.

Reply:
These are two different APIs and use different types.
It's already good that these two functions are next to each other in a file with nothing else.

Reply:
Oh right, I focused on `KOTLIN_1_8 and missed they are no on the same types ...